ui/src/seminars/containers/__tests__/SeminarStartDateFilterContainter.test.jsx

```diff
@@ -42,56 +39,48 @@ describe('SeminarStartDateFilterContainer', () => {
       </Provider>
     );
 
-    expect(wrapper.find(EventStartDateFilter)).toHaveProp({
-      selection: constants.START_DATE_ALL,
-    });
+    const childButton = within(
+      getByTestId('event-start-date-filter')
+    ).getByRole('switch');
+    expect(childButton).toHaveAttribute('aria-checked', 'false');
   });
 
-  it('dispatches SEARCH_QUERY_UPDATE onChange with start_date and sort=datedesc and empties timezone if all', () => {
+  it('flips button to all and then to upcoming', () => {
     const store = getStore();
     const namespace = SEMINARS_NS;
-    const wrapper = mount(
+    const { getByTestId } = render(
       <Provider store={store}>
         <SeminarStartDateFilterContainer
           namespace={namespace}
           switchTitle="title"
         />
       </Provider>
     );
-    const onChange = wrapper.find(EventStartDateFilter).prop('onChange');
-    onChange(constants.START_DATE_ALL);
 
-    const query = {
+    const quertAll = {
       start_date: constants.START_DATE_ALL,
       page: '1',
       sort: 'datedesc',
       timezone: undefined,
     };
-    const expectedActions = [searchQueryUpdate(namespace, query)];
-    expect(store.getActions()).toEqual(expectedActions);
-  });
-
-  it('dispatches SEARCH_QUERY_UPDATE onChange with start_date and sort=dateasc and empties timezone if upcoming', () => {
-    const store = getStore();
-    const namespace = AUTHOR_SEMINARS_NS;
-    const wrapper = mount(
-      <Provider store={store}>
-        <SeminarStartDateFilterContainer
-          namespace={namespace}
-          switchTitle="title"
-        />
-      </Provider>
-    );
-    const onChange = wrapper.find(EventStartDateFilter).prop('onChange');
-    onChange(constants.START_DATE_UPCOMING);
-
-    const query = {
+    const queryUpcoming = {
       start_date: constants.START_DATE_UPCOMING,
       page: '1',
       sort: 'dateasc',
       timezone: undefined,
     };
-    const expectedActions = [searchQueryUpdate(namespace, query)];
+    const expectedActions = [
+      searchQueryUpdate(namespace, quertAll),
+      searchQueryUpdate(namespace, queryUpcoming),
+    ];
+
+    const childButton = within(
+      getByTestId('event-start-date-filter')
+    ).getByRole('switch');
+    fireEvent.click(childButton);
```

Comment:
I would propose that we keep the same approach as before by splitting the tests in 2 as they were. One test will render and check that the default state is dispatched. One other will do the click and check that the other state is dispatched. I would avoid merging them in one.

Reply:
Agreed we can have one for the default state and one like you have it. In this one that you combine both , lets also leave a small comment for the the click actions so as to know from a first reading in which state you should be before performing the action.